packages/wrangler/src/pages/functions/routes-transformation.test.ts

```diff
@@ -0,0 +1,35 @@
+import { convertRoutePathsToGlobPatterns } from "./routes-transformation";
+
+describe("route-paths-to-glob-patterns", () => {
+  describe("convertRoutePathsToGlobPatterns()", () => {
+    it("should pass through routes with no wildcards", () => {
+      expect(convertRoutePathsToGlobPatterns(["/api/foo"])).toEqual(["/api/foo"]);
+      expect(convertRoutePathsToGlobPatterns(["/api/foo", "/api/bar"])).toEqual(["/api/foo", "/api/bar"]);
+      expect(convertRoutePathsToGlobPatterns(["/api/foo", "/api/bar/foo", "/foo/bar"])).toEqual(["/api/foo", "/api/bar/foo", "/foo/bar"]);
+    });
+
+    it("should escalate a single param route to a wildcard", () => {
+      expect(convertRoutePathsToGlobPatterns(["/api/:foo"])).toEqual(["/api/*"]);
+      expect(convertRoutePathsToGlobPatterns(["/api/foo/:bar"])).toEqual(["/api/foo/*"]);
+      expect(convertRoutePathsToGlobPatterns(["/bar/:barId/foo"])).toEqual(["/bar/*"]);
+      expect(convertRoutePathsToGlobPatterns(["/bar/:barId/foo/:fooId"])).toEqual(["/bar/*"]);
+      expect(convertRoutePathsToGlobPatterns(["/api/:foo", "/bar/:barName/profile", "/foo/bar/:barId/:fooId"])).toEqual(["/api/*", "/bar/*", "/foo/bar/*"]);
+    });
+
+    it("should pass through a single wildcard route", () => {
+      expect(convertRoutePathsToGlobPatterns(["/api/:baz*"])).toEqual(["/api/*"]);
+      expect(convertRoutePathsToGlobPatterns(["/api/foo/bar/:baz*"])).toEqual(["/api/foo/bar/*"]);
+      expect(convertRoutePathsToGlobPatterns(["/api/:foo/:bar*"])).toEqual(["/api/*"]);
+      expect(convertRoutePathsToGlobPatterns(["/foo/:foo*/bar/:bar*"])).toEqual(["/foo/*"]);
```

Comment:
This pattern *shouldn't* happen in the real world, but I'm glad we're testing this!

Reply:
yeah, I wouldn't it expect IRL either, but I figured better be safe than sorry